### ISSUE TITLE
Zone files not detected by plugin

### ISSUE THREAD

**Issue by sammcj** [sixty4k/st2-zonefile#14]
We have to manually set the syntax to Bind Zone File every time we open a zone file.

Is it possible to make this auto detect?

Perhaps it could be done with https://github.com/facelessuser/ApplySyntax ?

**Comment by sixty4k** [comment]
What file extension (if any) are you using for your zone files?

I'll look into ApplySyntax as well, not likely going to happen till towards the end of the month.

m.

**Comment by sammcj** [comment]
Hi @sixty4k, thanks for your reply.
bind's default is not to use any extension on it's zonefiles - which indeed is a bit strange but perhaps with applysyntax you could set the root directory of your zone file path and anything without an extension beyond that could get highlighting?

**Comment by sixty4k** [comment]
Well, 3 years later, I think https://github.com/sixty4k/st2-zonefile/pull/19 has the fix.  Will do some testing and verification hopefully later this week.

**Comment by sixty4k** [comment]
Folks, do you have zone files I can test with?  Privately or publicly.  It's 3 jobs from the place I wrote this for, and I don't have a collection of zone files anymore.

**Comment by sammcj** [comment]
I do - but not till Monday 

--
Sam McLeod

**Comment by cmur2** [comment]
For my shamelessly derived [Atom package](https://github.com/cmur2/language-zonefile) (I'm indebted to you) I have an [example zonefile](https://github.com/cmur2/language-zonefile/blob/master/example.zone) that sports most of the _mean_ features I came across.

**Comment by sammcj** [comment]
Crap I forgot to test this at work today.

Do you still need someone to test it with a range of varying zone files?

If so, give me a quick way to install your branch of the plugin and I’ll do it first thing tomorrow.

--
Sam McLeod

**Comment by sixty4k** [comment]
@sammcj back from vacation and finally seeing messages here, you can just check out the repo in `your <sublime folder>/packages/` the `zone_file_detection` branch.

I have a collection of zone files to play with, and will give @cmur2's zone file a whirl as well.

**Comment by sixty4k** [comment]
I need to play with st2, but st3 (with my standard set of plugins) definitely did not autodetect any of the first line patterns I tried.

I'll try and block some time to play with off the shelf copies of ST2 and ST3 with no packages (maybe I've got something installed that's breaking things).  Work is busy busy, so this is going to be low priority for a bit.  

here's the docs I've been trying to work to: http://www.sublimetext.com/docs/3/syntax.html#header
